rmw_opensplice_cpp/src/rmw_publisher.cpp

```diff
@@ -170,6 +170,13 @@ rmw_create_publisher(
 
   publisher->implementation_identifier = opensplice_cpp_identifier;
   publisher->data = publisher_info;
+  publisher->topic_name = reinterpret_cast<const char *>(
+    rmw_allocate(sizeof(char) * strlen(topic_name) + 1));
```

Comment:
I use `sizeof(char)` in combination with `malloc`-ish functions as a way to express I'm creating space for `char`'s. If you just use `1` it isn't very obvious what you're doing, and leaving out the `sizeof` is a bad habit which might show up when you really should be using it. I'd say leave the `sizeof(char)` and fix the `memcpy` to include it if you like.